### DISCUSSION TITLE
Üç asal sayı sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#266] [Soru Cevap]
Farklı üç adet 2 basamaklı asal sayımız var.
-Bu üç sayının ortalaması asal sayıdır.
-Her sayı çiftinin ortalaması asal sayıdır.

Bu üç sayıyı bulunuz.

**Comment by OmerB2** [comment]
```python
def asal(x):
    if x < 2:
        return False
    for i in range(2, int(x**0.5) + 1):
        if x % i == 0:
            return False
    return True

for i in range(11, 94):
    for x in range(11, 94):
        for y in range(11, 94):
            if asal(i) and asal(x) and asal(y):
                if i == x or x == y or i == y:
                    continue
                if asal((i + x + y) / 3) and asal((i + x) / 2) and asal((x + y) / 2) and asal((i + y) / 2):
                    print(i,x,y)
                else:
                    continue
            else:
                continue
```

**Reply by OmerB2** [reply]
Hocam doğru mu pek emin olamadım bir de birden fazla üçlü çıktı verdi ama benim kendi elimle test ettiğim üçlüler şartlara uygun çıktı.

**Reply by sonsuzus** [reply]
Çıktısına baktım da. 35 gibi 49 gibi asal olmayan sayılar da var?

**Reply by sonsuzus** [reply]
Ayrıca `asal((x + y) / 2) % 1 == 0` bu kod ne anlama geliyor? 

asal() fonksiyonu True veya False döndürüyor. Yani `True % 1 == 0 `mı diye bakıyorsun?

**Reply by OmerB2** [reply]
hocam `asal((x + y) / 2) % 1 == 0` kısmı yanlışlıkla olmuş diğer iften kopyala yapıştır yapmıştım o kısım aslında bölüm ondalıklı çıkıyor mu diye kontrol edecekti

**Reply by OmerB2** [reply]
hocam son yanlışı da düzelttim düzgün çalışması lazım.

**Reply by sonsuzus** [reply]
%1 ==0 her zaman doğru olur ki. Kodda mantıksal hatalar var.

**Reply by OmerB2** [reply]
hocam ondalıklı olduğunda bire bölümünden kalan sıfırdan büyük çıkmaz mı?

**Reply by sonsuzus** [reply]
Ondalıklı ifadelere bakma zaten hiç.

**Reply by OmerB2** [reply]
```python
def asal(x):
    if x < 2:
        return False
    for i in range(2, int(x**0.5) + 1):
        if x % i == 0:
            return False
    return True

for i in range(11, 94):
    for x in range(11, 94):
        for y in range(11, 94):
            if asal(i)==True and asal(x)==True and asal(y)==True:
                if i == x or x == y or i == y:
                    continue
                if asal((i + x + y) // 3) == True and asal((i + x) // 2) == True and asal((x + y) // 2) == True and asal((i + y) // 2) == True:
                    print(i,x,y)
                else:
                    continue
            else:
                continue
```
böyle mi hocam?

**Reply by sonsuzus** [reply]
Kodu kısaltabilirsin @OmerB2 

Örneğin 
`asal((i + x + y) // 3) == True`

yazmak yerine

`asal((i + x + y) /3)`

şeklinde yazabilirsin. Zaten True ise True dur.
